### DISCUSSION TITLE
Is it possible to "go to definition" for steps that come from an assembly within a nuget package?

### DISCUSSION THREAD

**Discussion by jokwodo** [reqnroll/Reqnroll#167] [Q&A]
I am currently in the middle of migrating a Specflow/Net6 project to Reqnroll/Net8. In addition to this migration, we are also needing to package up our step libraries into nuget packages. I have found this specifically has caused some issues with our framework, namely the following.

When I now attempt to F12/go to definition for a step contained within a nuget, it will now throw the following error within the output.

"Warning: ShowProblem: User Notification: Unable to jump to the step definition. No source location detected."

Is this a limitation of using steps that come from a nuget, or am I potentially doing something wrong here?

**Comment by gasparnagy** [comment]
The Visual Studio extension loads the source location from the PDB file using the [dnlib](https://www.nuget.org/packages/dnlib) library here: https://github.com/reqnroll/Reqnroll.VisualStudio/blob/main/Connectors/Reqnroll.VisualStudio.ReqnrollConnector/SourceDiscovery/DnLib/DnLibDeveroomSymbolReader.cs (We use dnlib v3.3.5 currently.)

Based on the error you mention I think it does not find the PDB file in the package. But it is also a good question what would be the case if it would find the PDB with the source path there. We would need to resolve it with Source Link and even in that case it would only be a Web URL that would need to be downloaded, etc. Pretty complex. I don't think there is a public API of VS that we could use to do all this task, so we would need to re-implement all the magic that VS does to be able to support this.

Do you have any suggestion on how this should work? Do you want this to work on packages that were built locally (so the path in the PDB exists)?

**Reply by jokwodo** [reply]
Ok, that would make sense as right now the PDB's are embedded into the nuget, but sounds like even if I built a new package with them as standalone files it would still have the same issue.

A fallback where if the projects are pulled down and built would work, but sounds like it would take a fair bit of effort, and not sure how that would work when sourcelink puts in the URL of the file on the repository.

**Reply by gasparnagy** [reply]
Not a solution, but I was thinking on providing better support for code documentation. If you would make some code comments on the step definition method, we could display those code comments in the editor if you hover over a defined step. So basically if you would want to make a reusable library of steps, you could make sure that those are properly documented in code. But this is just an idea still...
